internal/generator/vector/output/cloudwatch/cloudwatch_test.go

```diff
@@ -0,0 +1,221 @@
+	obs "github.com/openshift/cluster-logging-operator/api/observability/v1"
+	"github.com/openshift/cluster-logging-operator/internal/constants"
+	"github.com/openshift/cluster-logging-operator/internal/generator/framework"
+	. "github.com/openshift/cluster-logging-operator/internal/generator/vector/output/cloudwatch"
+	testhelpers "github.com/openshift/cluster-logging-operator/test/helpers"
+	. "github.com/openshift/cluster-logging-operator/test/matchers"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var _ = Describe("Generating vector config for cloudwatch output", func() {
+
+	const (
+		roleArn    = "arn:aws:iam::123456789012:role/my-role-to-assume"
+		secretName = "vector-cw-secret"
+	)
+	Context("#New", func() {
+
+		const (
+			groupPrefix           = "all-logs"
+			keyId                 = "AKIAIOSFODNN7EXAMPLE"
+			keySecret             = "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY" //nolint:gosec
+			secretName            = "vector-cw-secret"
+			secretWithCredentials = "secretwithcredentials"
+			vectorTLSSecret       = "vector-cw-secret-tls"
+		)
+
+		var (
+			tlsSpec = &obs.OutputTLSSpec{
+				CA: &obs.ConfigMapOrSecretKey{
+					Secret: &corev1.LocalObjectReference{
+						Name: vectorTLSSecret,
+					},
+					Key: constants.TrustedCABundleKey,
+				},
+				Certificate: &obs.ConfigMapOrSecretKey{
+					Secret: &corev1.LocalObjectReference{
+						Name: vectorTLSSecret,
+					},
+					Key: constants.ClientCertKey,
+				},
+				Key: &obs.SecretKey{
+					Secret: &corev1.LocalObjectReference{
+						Name: vectorTLSSecret,
+					},
+					Key: constants.ClientPrivateKey,
+				},
+				InsecureSkipVerify: false,
+			}
+
+			initOutput = func() obs.OutputSpec {
+				return obs.OutputSpec{
+					Type: loggingv1.OutputTypeCloudwatch,
+					Name: "cw",
+					Cloudwatch: &obs.Cloudwatch{
+						Region: "us-east-test",
+						Authentication: &obs.CloudwatchAuthentication{
+							AccessKeyID: &obs.SecretKey{
+								Secret: &corev1.LocalObjectReference{
+									Name: secretName,
+								},
+								Key: constants.AWSAccessKeyID,
+							},
+							AccessKeySecret: &obs.SecretKey{
+								Secret: &corev1.LocalObjectReference{
+									Name: secretName,
+								},
+								Key: constants.AWSSecretAccessKey,
+							},
+						},
+					},
+				}
+			}
+
+			secrets = map[string]*corev1.Secret{
+				secretName: {
+					Data: map[string][]byte{
+						constants.AWSAccessKeyID:     []byte(keyId),
+						constants.AWSSecretAccessKey: []byte(keySecret),
+					},
+				},
+				vectorTLSSecret: {
+					Data: map[string][]byte{
+						constants.ClientCertKey:      []byte("-- crt-- "),
+						constants.ClientPrivateKey:   []byte("-- key-- "),
+						constants.TrustedCABundleKey: []byte("-- ca-bundle -- "),
+					},
+				},
+				secretWithCredentials: {
+					Data: map[string][]byte{
+						constants.AWSCredentialsKey:  []byte("[default]\nrole_arn = " + roleArn + "\nweb_identity_token_file = /var/run/secrets/token"),
```

Comment:
I'm very confused by this (and see it in our 5.9 as well).   Why is there a role specified in a TLS secret?    Vector requires the env vars to be set in order to use the role.   I can't remember why it wouldn't work without the role and token path via ENV vars, but I'm fairly sure this TLS has anything to do with roles.  Am I confused?

Reply:
Conceptionally this is the same as before;  you spec the role and we read it from a secret and populate the env var.  This specific case is just convenience.  The new API allows for the admin to tell us which secret and key to us but from a config point of view, we don't care.  The tests were parsing a key out of these files are hard-coding them into the config